From: Bernhard Walle <bernhard@bwalle.de>
Date: Sat, 21 Jan 2012 23:48:27 +0100
Subject: [PATCH] Use own implementation of __rawmemchr()

rawmemchr() and its alias __rawmemchr() are non-standard function.
For compatibility with non-GNU platforms, use a self-implemented
_rawmemchr().

Signed-off-by: Bernhard Walle <bernhard@bwalle.de>
---
 eglibc/intl/explodename.c |    9 ++++++++-
 1 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/eglibc/intl/explodename.c b/eglibc/intl/explodename.c
index 321204d..c58abad 100644
--- a/eglibc/intl/explodename.c
+++ b/eglibc/intl/explodename.c
@@ -50,6 +50,13 @@ _nl_find_language (name)
   return (char *) name;
 }
 
+static char *_rawmemchr(const void *s, int c)
+{
+  char *p = (unsigned char *)s;
+  while (*p != c)
+    p++;
+  return p;
+}
 
 int
 _nl_explode_name (name, language, modifier, territory, codeset,
@@ -78,7 +85,7 @@ _nl_explode_name (name, language, modifier, territory, codeset,
   if (*language == cp)
     /* This does not make sense: language has to be specified.  Use
        this entry as it is without exploding.  Perhaps it is an alias.  */
-    cp = __rawmemchr (*language, '\0');
+    cp = _rawmemchr (*language, '\0');
   else if (cp[0] != '@')
     {
       if (cp[0] == '_')
